### PR TITLE
Add MVP media server

### PR DESCRIPTION
**Pull request by beardedfoo**
This MVP server is capable of serving MKV files to VLC

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+    def __init__(self, base_dir):
+        self.base_dir = base_dir
+        self.movies = {}
+
+    def scan(self):
+        movies = {}
+
+        for mkv_path in Path(self.base_dir).rglob('*.mkv'):
+            # Extract the MKV tags xml
+            extract_result = subprocess.run(['mkvextract', mkv_path.as_posix(), 'tags', '/dev/stdout'], capture_output=True)
```
Comment (maxpaj) [inline main.py:54]: I'd suggest adding something to the README about how to install these MKV tools.

https://mkvtoolnix.download/downloads.html 

If there will be an option to run this as a Docker container we could just put it on there.

File: main.py [hunk 2]
```diff
@@ -0,0 +1,420 @@
...
+
+    def do_GET(self):
+        self.method = 'GET'
+        return self.serve()
+
+    def do_HEAD(self):
+        self.method = 'HEAD'
+        return self.serve()
+
+    # TODO: Overwrite send_error to also send CORS headers
```
Comment (maxpaj) [inline main.py:129]: For something more than an MVP, I think flask or something would be nice.

File: main.py [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+
+        if self.method == 'GET':
+            try:
+                shutil.copyfileobj(f, self.wfile)
+            except ConnectionResetError:
+                self.log_error('connection reset by peer %s:%s', self.client_address[0], self.client_address[1])
+
+    def send_movie_genres_index(self):
+	# From https://api.themoviedb.org/3/genre/movie/list?api_key=<<api_key>>&language=en-US
+        genres = [
```
Comment (maxpaj) [inline main.py:248]: Do these ever change? We could just put them in a JSON file somewhere in the repo. What do you think about that?

File: main.py (hunk 1 above)
Comment (beardedfoo) [reply main.py:54]: Indeed the docker container is the current way. Once we get this merged in I can get the container building and on github images and then make a README mentioning the container image.

File: main.py (hunk 2 above)
Comment (beardedfoo) [reply main.py:129]: We can talk about this, but that's actually quite difficult given the level of control I need over headers and other finer details of the HTTP response. I tried building this using something like flask and ran into lots of problems with the framework messing up my responses. For the video streaming we will likely continue to need raw control over the responses.

File: main.py [hunk 4]
```diff
@@ -0,0 +1,420 @@
...
+
+            # Seek to the beginning of the range in the content
+            f.seek(range_start, SEEK_FROM_BEG)
+
+            # Set the approtriate code/headers for ranged content
+            self.send_response(HTTPStatus.PARTIAL_CONTENT)
+            self.send_header('Content-Range', f'bytes {range_start}-{total_size-1}/{total_size}')
+            self.send_header('Content-Length', str(total_size-range_start))
+        else:
+            # Seek to the beginning of the content since no range was specifieg
```
Comment (a-k-sea) [inline main.py:218]: Could you clarify this comment? There is a typo.

File: main.py (hunk 3 above)
Comment (beardedfoo) [reply main.py:248]: They do change (TMDB sets them). There's actually an API call you're meant to call to fetch these from TMDB but I hardcoded them to reduce calls upstream. I'm not opposed to dumping them in a JSON somewhere. I'll make an issue for it.

File: main.py [hunk 5]
```diff
@@ -0,0 +1,420 @@
...
+                shutil.copyfileobj(f, self.wfile)
+            except ConnectionResetError:
+                self.log_error('connection reset by peer %s:%s', self.client_address[0], self.client_address[1])
+
+    def send_movie_genres_index(self):
+	# From https://api.themoviedb.org/3/genre/movie/list?api_key=<<api_key>>&language=en-US
+        genres = [
+            {
+              "id": 28,
+              "name": "Action"
```
Comment (a-k-sea) [inline main.py:251]: Should we localize name at some point?

File: main.py [hunk 6]
```diff
@@ -0,0 +1,420 @@
...
+                break
+        else:
+            return self.send_error(HTTPStatus.NOT_FOUND)
+
+        # Extract the pertinent information from the metadata line
+        mime_type = m.group('mime_type')
+        attachment_id = m.group('attachment_id')
+        filename = m.group('filename')
+
+        # Extact the attachment content
```
Comment (a-k-sea) [inline main.py:391]: There is a typo in the comment.

File: main.py (hunk 1 above)
Comment (beardedfoo) [reply main.py:54]: https://github.com/duckflix/media-server/issues/2

File: main.py (hunk 3 above)
Comment (beardedfoo) [reply main.py:248]: https://github.com/duckflix/media-server/issues/3

File: main.py (hunk 2 above)
Comment (beardedfoo) [reply main.py:129]: https://github.com/duckflix/media-server/issues/4

File: main.py (hunk 5 above)
Comment (beardedfoo) [reply main.py:251]: https://github.com/duckflix/media-server/issues/5

File: main.py (hunk 6 above)
Comment (beardedfoo) [reply main.py:391]: https://github.com/duckflix/media-server/issues/5

File: main.py (hunk 4 above)
Comment (beardedfoo) [reply main.py:218]: https://github.com/duckflix/media-server/issues/5

File: main.py (hunk 5 above)
Comment (beardedfoo) [reply main.py:251]: https://github.com/duckflix/media-server/issues/6

**Pull request merged**